ntoskrnl/io/pnpmgr/pnpmgr.c

```diff
@@ -1841,11 +1940,15 @@ IopQueryCompatibleIds(PDEVICE_NODE DeviceNode,
       &Stack);
    if (NT_SUCCESS(Status) && IoStatusBlock.Information)
    {
-      /*
-      * FIXME: Check for valid characters, if there is invalid characters
-      * then bugcheck.
-      */
+      IsValidID = IopValidateID((PWCHAR)IoStatusBlock.Information, BusQueryCompatibleIDs);
+
+      if (!IsValidID)
+      {
+         DPRINT1("Invalid CompatibleIDs. DeviceNode - %p\n", DeviceNode);
```

Comment:
I suppose you should do more than that, either returning FALSE here, or bugchecking.

Reply:
I agree that it is necessary to do something, if the verification has not passed. It is possible to add DNF_HAS_PROBLEM or something else. But definitely not BugCheck.

Reply:
Ok, maybe try with another illegal character instead of just space. For example something larger than 0x7F (or strictly smaller than 0x20)

In case you make it bugcheck, here is the BSOD code that should be reported:
https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/bug-check-0xca--pnp-detected-fatal-error
Search for "Invalid ID" in there.

Reply:
I tested (usbhub_new) in Windows - bugchecking does not happen. If I add an error (except for a spaces) - a flash drive attached to the hub is not recognized. But Windows continues to work.

Reply:
I added ',' and '\'.